mlir-sycl/lib/Conversion/SYCLToLLVM/SYCLToLLVM.cpp

```diff
@@ -369,6 +369,43 @@ class ConstructorPattern final
   }
 };
 
+//===----------------------------------------------------------------------===//
+// MethodPattern - Converts SYCLMethodOpInterface to LLVM.
+//===----------------------------------------------------------------------===//
+
+static LogicalResult rewriteMethod(SYCLMethodOpInterface op,
+                                   ConversionPatternRewriter &rewriter) {
+  LLVM_DEBUG(llvm::dbgs() << "MethodPattern: Rewriting op: "; op.dump();
+             llvm::dbgs() << "\n");
+
+  SmallVector<mlir::Value> MethodArgs(op->getOperands());
+
+  if (op.getBaseType() != MethodArgs[0].getType()) {
+    auto Cast = rewriter.create<SYCLCastOp>(op.getLoc(), op.getBaseType(),
+                                            op->getOperand(0));
+    MethodArgs[0] = Cast;
+  }
+
+  rewriter.replaceOpWithNewOp<SYCLCallOp>(
```

Comment:
OK so is `SYCLCallOp` intended to be used as a target for each specialized SYCL method operation ?